src/Platform/Microsoft.Testing.Platform/PublicAPI/PublicAPI.Unshipped.txt

```diff
@@ -9,11 +9,22 @@ Microsoft.Testing.Platform.Extensions.Messages.FileArtifactProperty.FileArtifact
 Microsoft.Testing.Platform.Extensions.Messages.FileArtifactProperty.FileArtifactProperty(System.IO.FileInfo! FileInfo, string! DisplayName, string? Description = null) -> void
 Microsoft.Testing.Platform.Extensions.Messages.FileArtifactProperty.FileInfo.get -> System.IO.FileInfo!
 Microsoft.Testing.Platform.Extensions.Messages.FileArtifactProperty.FileInfo.init -> void
+Microsoft.Testing.Platform.Requests.AggregateFilter
+Microsoft.Testing.Platform.Requests.AggregateFilter.AggregateFilter(params System.Collections.Generic.IReadOnlyList<Microsoft.Testing.Platform.Requests.ITestExecutionFilter!>! innerFilters) -> void
+Microsoft.Testing.Platform.Requests.AggregateFilter.InnerFilters.get -> System.Collections.Generic.IReadOnlyList<Microsoft.Testing.Platform.Requests.ITestExecutionFilter!>!
+Microsoft.Testing.Platform.Requests.AggregateFilter.IsEnabled.get -> bool
+Microsoft.Testing.Platform.Requests.AggregateFilter.MatchesFilter(Microsoft.Testing.Platform.Extensions.Messages.TestNode! testNode) -> bool
+Microsoft.Testing.Platform.Requests.ITestExecutionFilter.IsEnabled.get -> bool
+Microsoft.Testing.Platform.Requests.ITestExecutionFilter.MatchesFilter(Microsoft.Testing.Platform.Extensions.Messages.TestNode! testNode) -> bool
```

Comment:
if we want to open up the filtering, we should have a discussion on whether this is the right abstraction.

- will it handle the tree filter correctly?
- some test frameworks have a notion of focus tests, where if any test has a focus label only that test is run. is this an example where the filter factory would need to query for all of the test nodes in the dll?
- longer term, can we optimize discovery via the filter? the current `MatchesFilter()` requires the framework to discover all test nodes. would we be able to optimize the discovery (to only discover tests in a class) if the test execution filters can be added externally as a delegate?

Reply:
> will it handle the tree filter correctly?
Why wouldn't it?

>  the current MatchesFilter() requires the framework to discover all test nodes

What do you mean? Currently it just accepts a single node, so it doesn't care about the rest of them.

> some test frameworks have a notion of focus tests, where if any test has a focus label only that test is run. is this an example where the filter factory would need to query for all of the test nodes in the dll?

I had an idea for a BatchFilter too. This would need a way of passing it all the test nodes so it has the information it needs.

Reply:
My initial idea was an interface like `IReceivesAllTestNodes` with a method that takes testnodes as a parameter, allowing the filter to then store them within a field or something.

This would then be up to the framework to call. Or alternatively, could a DataConsumer of DiscoveredTestNodes exist that aggregates them all and could pass them to the filter if it has that interface? Can a DataConsumer see a filter? 🤷‍♂️

Reply:
Actually - Could a filter itself just also be registered as a data consumer itself?

Reply:
I've added the ability for test filters to also be data consumers. See here: https://github.com/microsoft/testfx/pull/4200#discussion_r2035135071